cmd/cvp-trigger/main.go

```diff
@@ -4,69 +4,165 @@ import (
-	fs.StringVar(&o.prowConfigPath, prowConfigPathOption, "", "Path to the Prow config file")
-	fs.StringVar(&o.jobConfigPath, jobConfigPathOption, "", "Path to the Prow job config directory")
-	fs.StringVar(&o.periodic, "periodic", "", "Name of the Periodic job to manually trigger")
+	if o.bundleImageRef == "" {
+		return fmt.Errorf("required parameter %s was not provided", bundleImageRefOption)
+	}
 
-	_ = fs.Parse(os.Args[1:])
-	return o
-}
+	if o.channel == "" {
+		return fmt.Errorf("required parameter %s was not provided", channelOption)
+	}
 
-func (o options) validate() error {
-	if o.prowConfigPath == "" {
-		return fmt.Errorf("required parameter %s was not provided", prowConfigPathOption)
+	if o.indexImageRef == "" {
+		return fmt.Errorf("required parameter %s was not provided", indexImageRefOption)
 	}
 
 	if o.jobConfigPath == "" {
 		return fmt.Errorf("required parameter %s was not provided", jobConfigPathOption)
 	}
+	exists, _ := afs.Exists(o.jobConfigPath)
+	if !exists {
+		return fmt.Errorf("validating job config path %s failed, does not exist", o.jobConfigPath)
+	}
+
+	if o.jobName == "" {
+		return fmt.Errorf("required parameter %s was not provided", jobNameOption)
+	}
+
+	if o.ocpVersion == "" {
+		return fmt.Errorf("required parameter %s was not provided", ocpVersionOption)
+	}
+	if !strings.HasPrefix(o.ocpVersion, "4") {
+		return fmt.Errorf("ocp-version must be 4.x or higher")
+	}
 
-	if o.periodic == "" {
-		return fmt.Errorf("required parameter %s was not provided", periodicOption)
+	if o.packageName == "" {
+		return fmt.Errorf("required parameter %s was not provided", packageNameOption)
+	}
+
+	if o.prowConfigPath == "" {
+		return fmt.Errorf("required parameter %s was not provided", prowConfigPathOption)
+	}
+	exists, _ = afs.Exists(o.prowConfigPath)
+	if !exists {
+		return fmt.Errorf("validating prow config path %s failed, does not exist", o.prowConfigPath)
 	}
 
 	return nil
 }
 
+// getClusterConfig returns a config from calling util.LoadClusterConfig()
+func getClusterConfig() (*rest.Config, error) {
+	var clusterConfig *rest.Config
+	clusterConfig, err := util.LoadClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	return clusterConfig, nil
+}
+
+// getProwJob returns a Prow Job or an error if the provided job name is not found
+func getProwJob(jobName string, config *prowconfig.Config) (*pjapi.ProwJob, error) {
+	var selectedJob *prowconfig.Periodic
+	for _, job := range config.AllPeriodics() {
+		if job.Name == jobName {
+			selectedJob = &job
+			break
+		}
+	}
+
+	if selectedJob == nil {
+		return nil, fmt.Errorf("failed to find the job")
+	}
+
+	prowjob := pjutil.NewProwJob(pjutil.PeriodicSpec(*selectedJob), nil, nil)
+	return &prowjob, nil
+}
+
+// addEnvVarsToJobSpec adds the given key/value sets as EnvVars in the Env var list of the
+// .Spec.PodSpec.Containers[0].Env
+func addEnvVarsToJobSpec(job *pjapi.ProwJob, envVars map[string]string) {
```

Comment:
nit: functions with side effects present challenges in control flow and testing. Consider a non-mutating approach.

Reply:
Would the preference be to do this in line in `main()` instead?

Reply:
Or return a copy of the job that has the appended variables. If there are no other callers, though, I generally prefer to inline work (unless it's split out specifically to allow for unit testing)

Reply:
The reason I'd split out the inline calls were mainly to allow for unit testing. It doesn't seem like there's much of a focus on code coverage in the cmds I see here (not intended as a dig, at all, just to be clear). I generally try to keep `main()` primarily limited to logging and calls to other functions. I'm content to follow any conventions / standards, however, as there's a lot of existing go here.